src/shape.py

```diff
@@ -76,3 +76,65 @@ def create_shape(type: str, start_row: int, start_col: int) -> Shape:
         return SquareShape(start_row, start_col)
 
     raise NotImplementedError("Shape type not implemented")
+
+
+"""
+    U - - 
+    U - - 
+    U U -
+
+    U U U
+    U - -
+    - - -
+
+    - U U
+    - - U
+    - - U
+
+    - - -
+    - - U
+    U U U
+    """
+
+
+def rotate(shape: Shape) -> Shape:
+    unit_row_array = []
+    unit_col_array = []
+
+    for unit in shape.units:
+        unit_row_array.append(unit.row)
+        unit_col_array.append(unit.col)
+
+    row_diff = max(unit_row_array) - min(unit_row_array) + 1
+    col_diff = max(unit_col_array) - min(unit_col_array) + 1
+    min_value = min(min(unit_row_array), min(unit_col_array))
+    dimension = max(row_diff, col_diff)
+
+    grid = [[None for _ in range(dimension)] for _ in range(dimension)]
+    second_grid = [[None for _ in range(dimension)] for _ in range(dimension)]
```

Comment:
Bu tarz bir isim versek daha anlamli olabilir bunun icin
```suggestion
    rotated_grid = [[None for _ in range(dimension)] for _ in range(dimension)]
```